lib/chef/provider/windows_task.rb

```diff
@@ -297,6 +304,12 @@ def update_task_xml(options = [])
 
         doc = REXML::Document.new(xml_cmd.stdout)
 
+        if new_resource.frequency == :none
+          doc.root.elements.delete(xml_element_mapping["random_delay"])
+          cwd_element = REXML::Element.new(xml_element_mapping["random_delay"])
```

Comment:
Variable naming here is a bit unclear.  Why is this called `cwd_element`?

Also, can you explain what this code does, I feel like I'm missing some important detail.  It looks like we are deleting an element and then adding the same element right back.

Reply:
Thanks, this variable name should be `trigger_element` . I have made the changes.
As per approach explained in above comments,
This code is to delete value of start_time and startdate under `<TimeTrigger>`  from `<Triggers>` as below:
```
<Triggers>
    <TimeTrigger>
      <StartBoundary>2017-08-28T11:00:00</StartBoundary>
      <Enabled>true</Enabled>
    </TimeTrigger>
  </Triggers>
```
And then add only `<Triggers>` for frequncy `:none` as below:
` <Triggers />`